conda_vendor/conda_vendor.py

```diff
@@ -0,0 +1,330 @@
+import click
+import yaml
+import sys
+import struct
+import os
+import requests
+import hashlib
+import json
+from conda_vendor.version import __version__
+from conda_vendor.conda_lock_wrapper import CondaLockWrapper
+from conda_lock.src_parser import LockSpecification
+from conda_lock.conda_solver import DryRunInstall, VersionedDependency, FetchAction
+from pathlib import Path
+from typing import List
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+from conda_build import api
+from conda_vendor.iron_bank_generator import yaml_dump_ironbank_manifest 
+
+def get_lock_spec_for_environment_file(environment_file) -> LockSpecification:
+    lock_spec = CondaLockWrapper.parse_environment_file(environment_file)
+    return lock_spec
+
+
+# create the vendored channel directory, given the name in the environment.yaml
+def create_vendored_dir(environment_file, platform, desired_path=None) -> Path:
+    with open(environment_file, 'r') as env_file:
+        try:
+            environment_yaml = yaml.safe_load(env_file)
+            environment_name = environment_yaml['name']
+        except yaml.YAMLError as err:
+            click.echo(err)
+            sys.exit(f"Failed to read environment name from {environment_file}")
+    
+    # use current working directory if no path specified
+    if desired_path == None:
+        try:
+            path = os.path.join(os.getcwd(), environment_name)
+            os.mkdir(path)
+            create_platform_dir(path, platform)
```

Comment:
Would we not want to overwrite the current path by default ? if a user is rerunning ? or rather have the user delete it to avoid resolve.